FRENCH/src/ch17-02-trait-objects.md

```diff
@@ -0,0 +1,1021 @@
+        components: vec![
+            Box::new(SelectBox {
+                width: 75,
+                height: 10,
+                options: vec![
+                    String::from("Yes"),
+                    String::from("Maybe"),
+                    String::from("No")
+                ],
+            }),
+            Box::new(Button {
+                width: 50,
+                height: 10,
+                label: String::from("OK"),
+            }),
+        ],
+    };
+
+    screen.run();
+}
+```
+-->
+
+```rust,ignore
+use gui::{Ecran, Bouton};
+
+fn main() {
+    let ecran = Ecran {
+        composants: vec![
+            Box::new(ListeDeroulante {
+                largeur: 75,
+                hauteur: 10,
+                options: vec![
+                    String::from("Oui"),
+                    String::from("Peut-être"),
+                    String::from("Non")
+                ],
+            }),
+            Box::new(Bouton {
+                largeur: 50,
+                hauteur: 10,
+                libelle: String::from("OK"),
+            }),
+        ],
+    };
+
+    ecran.executer();
+}
+```
+
+<!--
+<span class="caption">Listing 17-9: Using trait objects to store values of
+different types that implement the same trait</span>
+-->
+
+<span class="caption">Encart 17-9 : Utilisation d'objets traits pour stocker des
+valeurs de types différents qui implémentent le même trait</span>
+
+<!--
+When we wrote the library, we didn’t know that someone might add the
+`SelectBox` type, but our `Screen` implementation was able to operate on the
+new type and draw it because `SelectBox` implements the `Draw` trait, which
+means it implements the `draw` method.
+-->
+
+Quand nous avons écrit la bibliothèque, nous ne savions pas que quelqu'un
+pourrait y ajouter le type `ListeDeroulante`, mais notre implémentation de
+`Ecran` a pu opérer sur le nouveau type et l'afficher parce que
+`ListeDeroulante` implémente le trait `Affichable`, ce qui veut dire qu'elle
+implémente la méthode `afficher`.
+
+<!--
+This concept—of being concerned only with the messages a value responds to
+rather than the value’s concrete type—is similar to the concept of *duck
+typing* in dynamically typed languages: if it walks like a duck and quacks
+like a duck, then it must be a duck! In the implementation of `run` on `Screen`
+in Listing 17-5, `run` doesn’t need to know what the concrete type of each
+component is. It doesn’t check whether a component is an instance of a `Button`
+or a `SelectBox`, it just calls the `draw` method on the component. By
+specifying `Box<dyn Draw>` as the type of the values in the `components`
+vector, we’ve defined `Screen` to need values that we can call the `draw`
+method on.
+-->
+
+Ce concept — se préoccuper uniquement des messages auxquels une valeur répond
+plutôt que du type concret de la valeur — est similaire au concept du *duck
+typing* (“typage canard”) dans les langages typés dynamiquement : si ça marche
```

Comment:
I would prefer keeping my translation as is since this concept is usually known as *duck typing* even for French speakers, and showing the French word-on-word translation is enough for the reader to understand the analogy.
(Besides, we really shouldn't rely on the readers actively searching a word in `translation-terms.md` to know the better-known English word; that file should primarily be aimed at the translators).